[BUG] Some fast parquet tests fail if the time zone is not UTC

**Describe the bug**
```
FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_written_by_spark_cpu[Timestamp(not_null)0][DATAGEN_SEED=0, INJECT_OOM] - AssertionError: GPU and CPU int values are different at [0, 'a']
FAILED ../../src/main/python/fastparquet_compatibility_test.py::test_reading_file_written_with_gpu[Timestamp(not_null)0][DATAGEN_SEED=0, INJECT_OOM] - AssertionError: GPU and CPU int values are different at [0, 'a']
```

```
SPARK_HOME=~/src/spark_3.4.2/ TEST_PARALLEL=0 DATAGEN_SEED=0 ./run_pyspark_from_build.sh -k 'test_reading_file_written_with_gpu or test_reading_file_written_by_spark_cpu'
```

Note that I am running on a computer with the timezone configured to 'America/Chicago' , but TZ itself is not set to anything.

```
$echo $TZ

```

If I set to `TZ='UTC'` or TZ='America/Chicago' then it works just fine, so I think this is a bug in how our test code detects time zones.  It is probably really minor.

--- a/integration_tests/run_pyspark_from_build.sh
+++ b/integration_tests/run_pyspark_from_build.sh
@@ -224,7 +224,7 @@ else
     fi
 
     # time zone will be tested; use export TZ=time_zone_name before run this script
-    TZ=${TZ:-UTC}
+    export TZ=${TZ:-UTC}
 
     # Disable Spark UI by default since it is not needed for tests, and Spark can fail to start
     # due to Spark UI port collisions, especially in a parallel test setup.

